app/js/components/MarginSelectorForm.jsx

```diff
@@ -0,0 +1,115 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {FormGroup, Label, InputGroup, InputGroupAddon, Alert} from 'reactstrap';
+import Input from 'react-validation/build/input';
+import {withNamespaces} from 'react-i18next';
+import {withRouter} from 'react-router-dom';
+import Form from 'react-validation/build/form';
+import {isInteger, required} from '../validators';
+
+class MarginSelectorForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      error: '',
+      marketError: '',
+      isAbove: props.margin.isAbove,
+      rate: props.margin.rate
+    };
+
+    // Redirecting to previous step
+    if(!props.fiat) this.props.history.push('/seller/fiat');
+  }
+
+  onRadioButtonChange = (e) => {
+    this.setState({isAbove: e.target.value === "above"});
+  }
+
+  onInputChange = (e) => {
+    this.setState({rate: e.target.value});
+  }
+
+  _submit(address){
+    if(!this.validate()) return;    
+    this.props.onSubmit(parseInt(this.state.rate, 10), this.state.isAbove); 
+    this.props.history.push(address);
+  }
+
+  back = () => {
+    this.props.history.push("/seller/fiat");
+  }
+
+  continue = () => {
+    // TODO: redirect to next step 
+    this._submit("/seller/margin");
+  };
+
+  validate() {
+    this.setState({error: '', marketError: ''});
+
+    this.form.validateAll();
+    if (this.form.getChildContext()._errors.length > 0) {
+      this.setState({error: this.props.t('marginSelectorForm.error')});
+      return false;
+    }
+
+    this.setState({error: ''});
+    return true;
+  }
+
+  render() {
+    const {t, fiat} = this.props;
+    const {error, rate, isAbove} = this.state;
+
+    if(!fiat) return null;
+
+    return ( 
+      <Form ref={c => { this.form = c; }}>
+        {error && <Alert color="danger">{error}</Alert>}
+        <FormGroup>
+            <Label for="rate">{t('marginSelectorForm.margin')}</Label>
+            <InputGroup>
+              <Input type="text" name="rate" id="rate" placeholder="0" className="form-control"
+                  value={rate} onChange={this.onInputChange} validations={[required, isInteger]} />
+              <InputGroupAddon addonType="append" style={{height: "calc(2.25rem + 2px)"}}>%</InputGroupAddon>
```

Comment:
Why do you need to specify the height manually? I thought it should use the same classes so be the same height by default.

Reply:
When a validation error for that field is shown, the InputGroupAddon extended vertically to the size of the input field + error message. I force the height to be a static value to avoid this.

Reply:
ooooh, ok thanks for the clarification. Maybe we could put that in `Form.scss` so that we don't have that problem with any other inputs